### ISSUE TITLE
TypeError: Cannot set property Headers of #<Object> which has only a getter w/ 0.0.138

### ISSUE THREAD

**Issue by Uninen** [vitest-dev/vitest#475]
### Describe the bug

Using Vitest v0.0.138 breaks with the following traceback when using with MirageJS / pretender:

```
TypeError: Cannot set property Headers of #<Object> which has only a getter
 ❯ node_modules/pretender/dist/pretender.js:1390:21
 ❯ node_modules/pretender/dist/pretender.js:1836:2
 ❯ Object.<anonymous> node_modules/pretender/dist/pretender.js:1850:3
 ❯ Object.<anonymous> node_modules/miragejs/dist/mirage-cjs.js:11:17
      9| var flatten = require('lodash.flatten');
     10| require('@miragejs/pretender-node-polyfill/before');
     11| var Pretender = require('pretender');
       |                 ^
     12| require('@miragejs/pretender-node-polyfill/after');
     13| var inflected = require('inflected');
```

The same code works with 137 and all previous versions at least down to ~120 or so.

### Reproduction

Install latest version of `miragejs` and set the following config so pretender dependency works:

```
    deps: {
      inline: ['pretender'],
    },
```

Then try to use Mirage in tests

```js
import { createServer } from 'miragejs'
```

### System Info

```shell
System:
    OS: macOS 12.1
    CPU: (8) arm64 Apple M1
    Memory: 333.45 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.13.1 - ~/.nvm/versions/node/v16.13.1/bin/node
    Yarn: 1.22.17 - /opt/homebrew/bin/yarn
    npm: 8.1.2 - ~/.nvm/versions/node/v16.13.1/bin/npm
  Browsers:
    Firefox: 95.0.2
  npmPackages:
    @vitejs/plugin-vue: ~2 => 2.0.1 
    vite: ~2.7 => 2.7.10 
    vitest: 0.0.138 => 0.0.138
```

### Used Package Manager

yarn

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by antfu** [comment]
How about this?

```ts  
deps: {
  inline: ['miragejs', 'pretender'],
},
```

**Comment by sheremet-va** [comment]
Can you confirm it still errors in the latest version?

**Comment by Uninen** [comment]
This has been fixed in v0.0.139 👍 

Thank You for very fast turnaround!
